samples/batch-process-document.v1beta3.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2020 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const uuid = require('uuid');
+
+async function main(
+  projectId = 'YOUR_PROJECT_ID',
+  location = 'YOUR_PROJECT_LOCATION',
+  processorId = 'YOUR_PROCESSOR_ID', // Create this in the Cloud Console
+  gcsInputUri = 'gs://cloud-samples-data/documentai/invoice.pdf',
+  gcsOutputUri = 'output-bucket',
+  gcsOutputUriPrefix = uuid.v4()
+) {
+  // [START documentai_batch_process_document]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = 'YOUR_PROJECT_ID';
+  // const location = 'YOUR_PROJECT_LOCATION'; // Format is 'us' or 'eu'
+  // const processorId = 'YOUR_PROCESSOR_ID';
+  // const gcsInputUri = 'YOUR_SOURCE_PDF';
+  // const gcsOutputUri = 'YOUR_STORAGE_BUCKET';
+  // const gcsOutputUriPrefix = 'YOUR_STORAGE_PREFIX';
+
+  // Imports the Google Cloud client library
+  const {
+    DocumentProcessorServiceClient,
+  } = require('@google-cloud/documentai').v1beta3;
+  const {Storage} = require('@google-cloud/storage');
+
+  const clientOptions = {
+    apiEndpoint: 'us-documentai.googleapis.com',
+  };
+
+  const client = new DocumentProcessorServiceClient(clientOptions);
+  const storage = new Storage();
+
+  async function batchProcessDocument() {
+    const name = `projects/${projectId}/locations/${location}/processors/${processorId}`;
+
+    const inputConfig = {
+      gcsSource: gcsInputUri,
+      mimeType: 'application/pdf',
+    };
+    const inputConfigs = [inputConfig];
+
+    // Configure the batch process request.
+    const request = {
+      name,
+      inputConfigs,
+      outputConfig: {
+        gcsDestination: `${gcsOutputUri}/${gcsOutputUriPrefix}/`,
+      },
+    };
+
+    // Batch process document using a long-running operation.
+    // You can wait for now, or get results later.
+    // Note: first request to the service takes longer than subsequent
+    // requests.
+    try {
+      const [operation] = await client.batchProcessDocuments(request);
+
+      // Wait for operation to complete.
+      await operation.promise();
+    } catch (ex) {
+      // CURRENTLY the LRO returned from service has 'done = true' but
```

Comment:
Heh, gross.  Can you add a link to the tracking bug with the upstream service here with a TODO so we know when it's safe to clean this up?

Reply:
To add to this point, in general I wouldn't recommend wrapping the sample in a try-catch block, it makes testing/debugging a bit harder sometimes. But,  I understand given the comment. Might be worth making a TODO comment to get rid of the block eventually?

Reply:
Question: would you consider this an issue with google-gax? Or should I pursue this issue with the Document AI team?

Reply:
After some investigation, I discovered that the API isn't sending the correct `Empty` response.

Googlers, see 170342661.

Reply:
This was marked as fixed on Oct 8th.  Can we drop this now?

Reply:
Done.